manager/usermanager/createUser.go

```diff
@@ -22,29 +20,25 @@ func CreateUser(userStore *store.UserStore, user *models.UserCredentials) (*mode
 		return nil, errors.ErrUserExists
 	}
 
-	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(user.GetPassword()), types.PasswordEncryptionCost)
+	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(user.Password), types.PasswordEncryptionCost)
 	if err != nil {
 		return nil, err
 	}
 
-	password := string(hashedPassword)
-	uid := uuid.Must(uuid.NewRandom()).String()
-
 	var newUser *models.UserCredentials
-	if user.GetRole() == models.RoleAdmin {
+	if user.Role == models.RoleAdmin {
 		newUser = user
-		newUser.Password = &password
+		newUser.Password = string(hashedPassword)
 	} else {
 		newUser = &models.UserCredentials{
-			UID:             &uid,
+			UID:             uuid.Must(uuid.NewRandom()).String(),
 			UserName:        user.UserName,
-			Password:        &password,
+			Password:        string(hashedPassword),
 			Name:            user.Name,
 			UnverifiedEmail: user.UserName,
 			Kind:            models.LocalAuth,
 			Role:            models.RoleUser,
 			OnBoardingState: models.BoardingStateSignup,
-			CreatedAt:       &time.Time{},
```

Comment:
Why is this removed?

Reply:
We are having `CreatedAt` field being set at the time of inserting the user https://github.com/mayadata-io/kubera-auth/blob/master/pkg/store/user.go#L91.
This was used for initializing the field but as we are taking the address there, it was not reqiured

Reply:
Got it !!